src/mqtt.c

```diff
@@ -238,6 +238,20 @@ static void on_message(
   sfree(vl.values);
 } /* void on_message */
 
+static int mqtt_subscribe(mqtt_client_conf_t *conf) {
+  int status;
+
+  status = mosquitto_subscribe(conf->mosq, /* message_id = */ NULL, conf->topic,
+                               conf->qos);
+  if (status != MOSQ_ERR_SUCCESS) {
+    ERROR("mqtt plugin: Subscribing to \"%s\" failed: %s", conf->topic,
+          mosquitto_strerror(status));
+    return -1;
```

Comment:
Since `MOSQ_ERR_SUCCESS == 0`, we can just do `return status;` below.

Reply:
Well, that's mixing two return value classes: 0 / -1 and mosquitto error codes.

From just reading of this code it isn't clear that `MOSQ_ERR_SUCCESS` means zero (one has to dig into a mosquitto header file for this), so for readability and future ABI changes compatibility all `mqtt_subscribe()` callers (two) would need to check for this particular constant too.

So it's better to "translate" it immediately to zero in the single place in `mqtt_subscribe()` so callers can use the 0 / -1 status values as in the most other places in this code file.